### PR TITLE
Add setting for forcing refresh in zypper

### PR DESCRIPTION
**Pull request by agraul**
### What does this PR do?
Currently `salt '*' pkg.refresh_db` will always call `zypper refresh` with `--force` when `zypper` is used as the backend. With this PR, this behavior can be controlled. It leaves the default as it is, but allows a CLI flag and/or a pillar entry to control if a repository is refreshed with `--force`.

### Previous Behavior
`salt '*' pkg.refresh_db` -> `zypper refresh --force`

### New Behavior
`salt '*' pkg.refresh_db` -> `zypper refresh --force`
`salt '*' pkg.refresh_db force=true` -> `zypper refresh --force`
`salt '*' pkg.refresh_db force=false` -> `zypper refresh`

Alternatively, a pillar setting can be used (which is overridden by the CLI option):
```yml
zypper:
  refreshdb_force: false
```

### Tests written?

Yes

### Commits signed with GPG?

Yes

### CODE REVIEW

**Review by brejoc — APPROVED** [review]
~~lgtm!~~ 

Sorry, had to revert my initial review.

File: salt/modules/zypperpkg.py [hunk 1]
```diff
@@ -1171,22 +1171,39 @@ def mod_repo(repo, **kwargs):
...
     # Remove rtag file to keep multiple refreshes from happening in pkg states
     salt.utils.pkg.clear_rtag(__opts__)
     ret = {}
-    out = __zypper__.refreshable.call('refresh', '--force')
+    refresh_opts = ['refresh']
+    if force is None:
+        force = __pillar__.get('zypper', {}).get('refreshdb_force', True)
+    if force:
+        refresh_opts.append('--force')
+    out = __zypper__(root=root).refreshable.call(*refresh_opts)
```
Comment (brejoc) [inline salt/modules/zypperpkg.py]: Oh, wait a sec. I think the root kwarg is not yet introducted here. Could you check that please?

**Review by brejoc — CHANGES_REQUESTED** [review]

File: salt/modules/zypperpkg.py (hunk 1 above)
Comment (agraul) [reply salt/modules/zypperpkg.py]: You are correct, that will be introduced with https://github.com/saltstack/salt/pull/54954

**Review by brejoc — DISMISSED** [review]
:+1:

**Review by brejoc — APPROVED** [review]

**Review by dwoz — APPROVED** [review]

**Pull request merged**